subsys/net/lib/fota_download/src/fota_download.c

```diff
@@ -133,6 +141,38 @@ int fota_download_start(char *host, char *file)
 		return -EINVAL;
 	}
 
+#ifdef PM_S1_ADDRESS
+	/* B1 upgrade is supported, check what B1 slot is active,
+	 * (s0 or s1), and update file to point to correct candidate if
+	 * space separated file is given.
+	 */
+	char *update;
+	struct fw_firmware_info s0;
+	struct fw_firmware_info s1;
+
+	err = spm_firmware_info(PM_S0_ADDRESS, &s0);
+	if (err != 0) {
+		return err;
+	}
+
+	err = spm_firmware_info(PM_S1_ADDRESS, &s1);
+	if (err != 0) {
+		return err;
+	}
+
+	bool use_s1 = s1.firmware_version > s0.firmware_version;
```

Comment:
This isn't always valid. If the version number of s1 is higher, but the signature check of s1 failed, this gives the wrong answer.

Reply:
Hm, so we need to do a prevalidate at this point? Would it be better for B0 to reset the version of non-accepted updates?